_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+
+Like any cross-platform abstraction, React Native can be leaky. To write a cross-platform app that purely lives inside JS Runtime, you have to write React-only code. React and React Native doesn't have ways to handle primitives like `UINavigationController` - they want your entire app to be represented as a series of components that can be mapped across many platforms. 
+
+This isn't optimal when you're coming in from the native world - where you're used to building platform-specific experiences, and are genuinely excited at the prospect of platform-specific APIs. Generally you can look for other teams who have felt the same and are willing to write native-bridged code that's specific to iOS. Shout-out to [Wix][] and [AirBnB][] who are doing great work in this space.
+
+Is this a critical problem against React Native? I don't think so, we've added native abstractions where it was the right decision and we've used JavaScript when it was the right decision. For example, our `Image` component is a bridged native component that uses `SDWebImage` under the hood so that we can share an image cache with the rest of the app.
+
+## Ten minutes to try out React
+
+OK, no joke, don't skip this bit, you try React Native right now. This will require some terminal skills, and about 5 minutes, it shouldn't be more complex than using CocoaPods via the terminal.
+
+</article>
+<article class="desktop-only" style='display: flex; flex-flow:row;'>
+<div style='flex:1; display: block;'>
+
+```sh
+# If you don't have homebrew
+# see https://brew.sh
+
+# Install the JavaScript tools you'll need
+brew install nodejs yarn
+
+# You'll want Visual Studio Code to work on this
+brew cask install visual-studio-code
+
+# Install the React Native CLI
+yarn global add react-native-cli
+
+# Create a new React Native
+# project called TrendingArtists
+react-native init TrendingArtists
+```
+
+</div>
+<div class="desktop-only" style='flex:1; display: block;'>
+
+<p>You'll need <code>node</code> and <code>yarn</code> installed globally so you can run JavaScript and handle dependency management respectively.<p>
+
+<p>For working inside a JavaScript project, I'd strongly recommend using Microsoft's <a href='https://code.visualstudio.com'>Visual Studio Code</a>, it does a great job for React Native.</p>
+
+<p>Next up we're going to make the initial project and look around, so once all the installing has finished. You can follow along with the next section.</p>
+
+</div>
+</article>
+<article class='post'>
+
+Alright, so that should do a lot of downloading, and you'll have a new folder with a fully set up project for iOS and Android. We'll be focusing on the iOS side.
+
+So, open up `TrendingArtists` in your editor and inside your terminal with `cd TrendingArtists`. From the terminal you can get the Xcode project compiled, and opened inside the iOS simulator with `react-native start ios`. 
```

Comment:
It wasn't immediately obvious to me that `react-native start ios` wouldn't open Xcode for me; I thought it had hung in the terminal. Being extra explicit in these steps could help ward off anyone getting stuck and frustrated.